app/views/charges/downgrade.html.erb

```diff
@@ -0,0 +1,4 @@
+<% if current_user.has_role? :premium %>
+    <h4>Click to downgrade to Standard user!</h4>
+    <%= link_to "Downgrade", method: :downgrade, class: 'btn btn-danger', data: { confirm: 'Are you sure you want to downgrade to standard user and relinquish power?' } %>
```

Comment:
@praveenyr I think this is where our wires are crossed -- the code you wrote is already invoking it since you are POSTing with the DELETE method. If you remove the get route and downgrade.html.erb, your code is perfect. I'm headed to roller derby practice now but will check in about 4 hours.